src/bin/stratisd.rs

```diff
@@ -78,6 +86,33 @@ fn handle_udev_add(event: &libudev::Event) -> Option<(PathBuf, Device)> {
     None
 }
 
+/// To ensure only one instance of stratisd runs at a time, acquire an
+/// exclusive lock. Exit if lock attempt fails.
+fn create_pid_file() -> StratisResult<File> {
+    let mut f = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(true)
+        .open(STRATISD_PID_PATH)?;
+    match flock(f.as_raw_fd(), FlockArg::LockExclusiveNonblock) {
+        Ok(()) => {f.write(format!("{}\n", getpid()).as_bytes())?;
+                  Ok(f)}
+        Err(_) => {
+            let mut buf = String::new();
+            f.read_to_string(&mut buf)?;
+            // pidfile is supposed to contain pid of holder. But you never
+            // know so be paranoid.
+            let pid_str = buf.split_whitespace()
+                .next()
+                .and_then(|s| s.parse::<pid_t>().ok())
+                .map(|pid| format!("{}", pid))
+                .unwrap_or_else(|| "<unknown>".into());
+            eprintln!("Stratisd is already running with pid {}", pid_str);
+            exit(1);
```

Comment:
Just return an error with a message. It will be printed out by the ```main()``` method, see below.